lib/types.ts

```diff
@@ -55,8 +53,25 @@ export interface ScannedProjectCustom {
    * a future change should be mindful but find a way to unify them if possible.
    */
   depTree: DepTree | PkgTree;
-  targetFile?: string;
-  meta?: any;
+  targetFile?: string; // currently used for application-dependencies scans
+  meta?: any; // not to pollute with actual data; reserved for actual metadata
+}
+
+enum ScanType {
+  DependencyTree,
+  DependencyGraph,
+  ManifestFiles,
+}
+
+export interface ScannedProjectExtended extends ScannedProjectCustom {
```

Comment:
WDYT of naming `ScannedProjectCustom` just `ScannedProject`? And then this new type `ScannedProjectExtended` could become `ScannedProjectCustom` or keep it `ScannedProjectExtended`

Reply:
```ScannedProjectCustom``` is used to reflect exactly what the CLI wants right now, I'd rather keep it this way
happy to rename ```ScannedProjectExtended``` to whatever makes sense to you